The job **`tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/19752744048/job/56598619463) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/56598619463) [(plain)](https://github.com/rust-lang/rust/commit/84ca72e5c143e5c7ee653265dca33ae335e66bb4/checks/56598619463/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
[TIMING:end] tool::Tidy { compiler: Compiler { stage: 0, host: x86_64-unknown-linux-gnu, forced_compiler: false }, target: x86_64-unknown-linux-gnu } -- 0.000
fmt check
fmt: checked 6576 files
tidy check
tidy [target-specific-tests (tests)]: /checkout/tests/assembly-llvm/stack-protector/stack-protector-safe-stack.rs: revision [unspecified] should specify `needs-llvm-components: x86` as it has `--target` set
tidy [target-specific-tests (tests)]: FAIL
tidy [rustdoc_json (src)]: `rustdoc-json-types` modified, checking format version
tidy: Skipping binary file check, read-only filesystem
removing old virtual environment
creating virtual environment at '/checkout/obj/build/venv' using 'python3.10' and 'venv'
creating virtual environment at '/checkout/obj/build/venv' using 'python3.10' and 'virtualenv'
---
linting javascript files and applying suggestions
Running eslint on rustdoc JS files
info: ES-Check: there were no ES version matching errors!  🎉
typechecking javascript files
tidy: The following check failed: target-specific-tests (tests)
Command `/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools-bin/rust-tidy /checkout /checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo /checkout/obj/build 4 /node/bin/yarn --extra-checks=py,cpp,js,spellcheck` failed with exit code 1
Created at: src/bootstrap/src/core/build_steps/tool.rs:1612:23
Executed at: src/bootstrap/src/core/build_steps/test.rs:1317:29

Command has failed. Rerun with -v to see more details.
Bootstrap failed while executing `test src/tools/tidy tidyselftest --extra-checks=py,cpp,js,spellcheck`
Build completed unsuccessfully in 0:02:53
  local time: Fri Nov 28 03:02:22 UTC 2025
  network time: Fri, 28 Nov 2025 03:02:23 GMT
##[error]Process completed with exit code 1.
```

</details>

--- a/tests/assembly-llvm/stack-protector/stack-protector-safe-stack.rs
+++ b/tests/assembly-llvm/stack-protector/stack-protector-safe-stack.rs
@@ -1,0 +1,54 @@
+//@ add-minicore
+//@ revisions: all strong none safestack safestack_strong safestack_all
+//@ assembly-output: emit-asm
+//@ only-x86_64-unknown-linux-gnu
+//@ [all] compile-flags: -Z stack-protector=all
+//@ [strong] compile-flags: -Z stack-protector=strong
+//@ [none] compile-flags: -Z stack-protector=none
+//@ [safestack] compile-flags: -Z stack-protector=none -Z sanitizer=safestack
+//@ [safestack_strong] compile-flags: -Z stack-protector=strong -Z sanitizer=safestack
+//@ [safestack_all] compile-flags: -Z stack-protector=all -Z sanitizer=safestack
+//@ compile-flags: -C opt-level=2 -Z merge-functions=disabled --target x86_64-unknown-linux-gnu
+//@ needs-llvm-components: x86
+
+#![feature(unsized_fn_params)]
+#![crate_type = "lib"]
+#![feature(no_core)]
+#![no_core]
+#![no_std]
+
+extern crate minicore;
+use minicore::*;
+
+extern "C" {
+    fn test4spss(p: *mut u8);
+}
+
+// CHECK-LABEL: test1{{:|\[}}
+#[no_mangle]
+pub unsafe fn test1(x: *mut u8) -> u8 {
+    let mut buf: [u8; 64] = [0; 64];
+
+    let p = &mut buf as *mut [u8; 64] as *mut u8;
+
+    *p = 42;
+
+    test4spss(p);
+
+    *x = *p;
+
+    *p
+
+    // none-NOT: __stack_chk_fail
+    // strong: __stack_chk_fail
+    // all: __stack_chk_fail
+
+    // safestack: __safestack_unsafe_stack_ptr
+    // safestack-NOT: __stack_chk_fail
+
+    // safestack_strong: __safestack_unsafe_stack_ptr
+    // safestack_strong: __stack_chk_fail
+
+    // safestack_all: __safestack_unsafe_stack_ptr
+    // safestack_all: __stack_chk_fail
+}

